### ISSUE TITLE
Show better errors for naming collisions when embedding

### ISSUE THREAD

**Issue by wolfgangwalther** [PostgREST/postgrest#2529]
Against the current spec test schema:

```
GET /organizations?select=id,referee(id,name),referee(id,name)
{"code":"42712","details":null,"hint":null,"message":"table name \"organizations_referee_1\" specified more than once"}
```

```
GET /organizations?select=id,organizations!referee(id,name),organizations!auditor(id,name)
{"code":"42803","details":null,"hint":null,"message":"aggregate functions are not allowed in FROM clause of their own query level"}
```

While the first one is kind of ok, because we can guess what the problem is, the second one is not.

We should error out when the select spec we have results in two columns with the same name instead of trying to build a query.

**Comment by steve-chavez** [comment]
I think it'd be better to err and prevent confusions. Ignoring already caused problems as seen on https://github.com/PostgREST/postgrest/issues/1771 (wrong queries can remain unnoticed)

> Maybe it would be more consistent to just throw away duplicate values, similar to how query strings are parsed in general.
> GET /rpc/getitemrange?min=1&min=2&max=4&max=5
> This will use min=2&max=5 only.

This is usual behavior on web applications AFAICT but our `select` param doesn't have to follow that - we can make it strict.

**Comment by steve-chavez** [comment]
Yes, agree. It should be done at the planner stage. Maybe by handling that the name or alias of the node is unique on the select tree.

**Comment by steve-chavez** [comment]
Bumped into this error when trying to use the new spread-to-many:

```bash
curl 'localhost:3000/titles?select=people!title_directors(director:primary_name),people!title_cast(actor:primary_name)'

{"code":"42803","details":null,"hint":null,"message":"aggregate functions are not allowed in FROM clause of their own query level"}
```

Note that the embedded table name should not matter here, as it shouldn't be included in the end result.
